core/test/java/game/providers/FreedomLineProviders.java

```diff
@@ -31,47 +58,109 @@ public class FreedomLineProviders {
     public static @NotNull Stream<Arguments> provideSetOfPositions() {
         return Stream.of(
                 Arguments.of(
-                        new TreeSet<>(Set.of(
+                        new LinkedHashSet<>(List.of(
```

Comment:
Why are you providing a set of lists? Wouldn't be `Set.of(...)` more than enough? Considering the `addPositions()` method only requires a `Set`

Reply:
Nope, because I need that the input positions are insertion ordered.

Reply:
But then the `addPositions()` method signature is incorrect! The method needs a `SortedSet` in order to work properly

Reply:
Note that with your last commit the test coverage is not more 100%

Reply:
Then it should be clearer what the method does: when does it fail? When does it succeed? Why does the test pass with ordered positions, but fail with unordered ones?

Reply:
I will create for each direction a different method provider in order to improve the readability.  Hope it helps